### PR TITLE
AAP-70872 Prefer visible JT in deprecated named URL reference

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
For case of

GET `/api/v2/job_templates/jt_name/`

where the name of the job template is "jt_name", this will now give different behavior. Where job templates both visible and non-visible to that user exists.

Before, if a non-visible JT was returned as the `.first()` in the query, the page would attempt to render but give a 403. In this revision, it will get the first visible JT, so this will change to a 200, giving the _other_ JT that the user has access to see.

Ambiguous cases still exist, which is why that URL pattern is deprecated in favor of `/api/v2/job_templates/org_name++jt_name/`

##### ISSUE TYPE
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
 - API

---

> [!NOTE]
> **Medium Risk**
> Changes request URL rewriting for deprecated job template named URLs to depend on the authenticated user’s visible queryset, which can alter which resource is returned (or whether it rewrites at all). Moderate risk because it touches middleware routing and permission-sensitive resolution paths.
> 
> **Overview**
> Adjusts `URLModificationMiddleware` so deprecated `/api/v2/job_templates/<name>/` named-URL rewriting resolves job templates using the *request user’s* permitted queryset (when authenticated) instead of an unscoped DB lookup, preventing rewrites to inaccessible templates and preferring a visible template when names collide.
> 
> Adds functional tests covering no-access, duplicate-name visibility selection, per-user resolution, fallback behavior when no user is provided, and org role edge cases for this deprecated URL form.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cfb4ba3f852fe742ff067d2c241b56da5483da88. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Enhanced job template URL resolution to be permission-aware, enabling users to be correctly routed to templates they can access when multiple templates share the same name across organizations.

* **Tests**
  * Added test coverage for permission-based job template URL resolution, including multi-user and multi-organization scenarios.

### CODE REVIEW

File: awx/main/middleware.py [hunk 1]
```diff
@@ -172,7 +174,7 @@ def _convert_named_url(cls, url_path):
 
     def process_request(self, request):
         old_path = request.path_info
-        new_path = self._convert_named_url(old_path)
+        new_path = self._convert_named_url(old_path, user=getattr(request, 'user', None))
```
[the remark this answers is not in the document]
Comment (AlanCoding) [reply awx/main/middleware.py:177]: oh this might be a good point

**Comment by AlanCoding** [comment]
abandoning for now
